.eslintrc.cjs

```diff
@@ -78,16 +78,16 @@ module.exports = {
       // Tighten rules for exported code.
       files: ['packages/*/src/**/*.js'],
       rules: {
-        // The rule is “no nested awaits” but the architectural goal is
-        // “no possibility of ‘awaits sometimes but not always’”. That is our
-        // architectural rule. If it’s too constraining you have to fall back to
-        // promise.then or get a reviewed exception.  “sometimes awaits” is a
-        // bug farm for particularly pernicious bugs in which you can combine
-        // two correct pieces of code to have emergent incorrect behavior.
-        // It’s absolutely critical for shared service code. That means
-        // contracts, but it also means kernel components that are used by
-        // multiple clients. So we enable it throughout the repo and exceptions
-        // are code-reviewed.
+        // The rule is “no nested awaits” but the architectural goal is “no
+        // possibility of ‘awaits sometimes but not always’”. That is our
+        // architectural rule for vats which require distributed determinism. If
```

Comment:
How is distributed determinism relevant? I thought the issue was correctness.

Reply:
Thanks for reviewing this part I forgot to mention. I was trying to distinguish why CLI and solo didn't need this linking.

My understanding is it's only relevant when there things can happen in different turns, ergo blockchain context.

What's a better description of when the lint rule needs to be enforced?

Reply:
What are the purposes of the `cli` and `solo` packages? What are the worst case consequences of an interleaving bug in the code of these packages?

Depending on the answers, we may want to leave these warnings on for these packages and carry through the triage for the problematic awaits is these packages.

Reply:
Different turns happen any time code doesn't complete sequentially; every `await` or `promise.then(...)` or I/O callback introduces another turn. No blockchain context needed.

My understanding is that the rule is relevant any time we want correct code; that is: to avoid a pattern that is known to cause bugs. I could be wrong. @erights ? @dtribble ?

Reply:
> What are the purposes of the cli and solo packages? What are the worst case consequences of an interleaving bug in the code of these packages?

CLI is ergonomics for an end-user (or dev) to craft and send commands. Solo is is the solo Vat runner, which tmk is for testing. I can't think of any dire consequences to bugs in either.

> Different turns happen any time code doesn't complete sequentially

Where can I read more about this? I've never encountered the term before Agoric and can't find it in web search.

If this PR needs to land for PS0, @erights feel free to remove any of my changes. I can follow up with them later.

Reply:
> I can't think of any dire consequences to bugs in either.

I think this is the important bit. It means the warnings should stay on, but fixing those is much less urgent. Perhaps those should only be classified "bug" without "security"?

Reply:
> Where can I read more about this? I've never encountered the term before Agoric and can't find it in web search.

https://papers.agoric.com/papers/concurrency-among-strangers/abstract/ , coauthored by myself, @dtribble , and Jonathan Shapiro.

This paper expanded into Part III of https://papers.agoric.com/papers/robust-composition/abstract/ , though the briefer standalone paper is certainly adequate for present purposes. I recommend you read my whole thesis in any case ;)

Reply:
Though they don't seem to use the term "turn", the concept is well known in the JS world, yes? A "turn" is just one turn through the event loop.

For example, https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop:

> At some point during the event loop, the runtime starts handling the messages on the queue, starting with the oldest one. To do so, the message is removed from the queue and its corresponding function is called with the message as an input parameter. As always, calling a function creates a new stack frame for that function's use.
>
> The processing of functions continues until the stack is once again empty. Then, the event loop will process the next message in the queue (if there is one).

Reply:
p.s. I do agree that the whole "Robust Composition" thesis is worth reading in due course. Though it's over 200 pages, it's not dense material. Much of it is stuff you almost certainly already know.

Reply:
> Solo is is the solo Vat runner, which tmk is for testing.

Solo powers `ag-solo`; it's it's intended, in due course, for dapp back-ends that may hold significant assets. @dtribble recently gave an example of a gateway to an online game world, I think.

Reply:
sounds like we should just drop my two eslint/comment changing commits. I figure that's easiest to do by interactive rebase and force push. To avoid clobbering any concurrent work I'll refrain unless you request it @erights

Reply:
I'll do it when I get back to working on this PR. Thanks.